ls1intum/Artemis
# Exam mode/client/student overview

### Checklist

### Motivation and Context


The students need an overview in order to see which exams are available.
They need to be routed to the correct view of the exam.

### Description

Exams are shown in the course overview.
Added logic which component should be displayed in the participation component.
Initialized all components for the exam views.
If there are no exams or no visible exams the 'exam' tab is not shown. 
If the exam is not visible it is not shown in the overview.
If an exam it visible and not yet over the start component is displayed when navigating to the exam.
If an exam is visible an over the summary component is displayed when navigating to the exam.

### Steps for Testing


*should be tested locally until exams can be created*
Setup
1. Connect to the Artemis database with tablePlus (or whichever tool you prefer)
2. Create a few exams and assign them to a course to play around with

Testing
1. Log in to Artemis
2. Navigate to the course with the exams
3. Check that the exams are displayed correctly
4. Click on the exam and check that the correct component is shown

### Screenshots


<img width="738" alt="Bildschirmfoto 2020-06-12 um 09 29 39" src="https://user-images.githubusercontent.com/51987021/84499673-55fd2b80-acb3-11ea-9a9b-ab5b33e1205a.png">
